src/Smalot/PdfParser/Element/ElementXRef.php

```diff
@@ -56,6 +56,24 @@ public function getObject()
      */
     public function equals($value)
     {
+        /**
+         * In case $value is a number and $this->value is a string like 5_0
+         *
+         * Without this if-clause code like:
+         *
+         *      $element = new ElementXRef('5_0');
+         *      $this->assertTrue($element->equals(5));
+         *
+         * would fail (= 5_0 and 5 are not equal in PHP 8.0+).
+         */
+        if (
+            true === is_numeric($value)
+            && true === \is_string($this->getContent())
+            && 1 === preg_match('/[0-9]+\_[0-9]+/', $this->getContent(), $matches)
+        ) {
+            return (float) ($this->getContent()) == $value;
+        }
+
```

Comment:
@smalot Do you see any issues with that addition?

Reply:
Can you remind me when does this method is called and why we try to validate equality between "5" and "5_0" ?

Reply:
If you remove this code, [assertion](https://github.com/smalot/pdfparser/blob/php8/tests/Integration/Element/ElementXRefTest.php#L115) in `testEquals` of ElementXRefTest.php will fail.